### PR TITLE
Build with -Wvla-cxx-extension

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

oss-fuzz build fails because of it.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit f4d4ef9dbfcfe6ffe61e7969bbb031ef38f26a82 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73119/f4d4ef9dbfcfe6ffe61e7969bbb031ef38f26a82/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by nickitat** [comment]
`Integration tests (aarch64) [3/6]` - #73113
`Integration tests (aarch64) [4/6]` - #73131

File: src/AggregateFunctions/AggregateFunctionHistogram.cpp [hunk 1]
```diff
@@ -130,10 +130,9 @@ class AggregateFunctionHistogramData
 
         // Maintain doubly-linked list of "active" points
         // and store neighbour pairs in priority queue by distance
-        UInt32 previous[size + 1];
-        UInt32 next[size + 1];
-        bool active[size + 1];
-        std::fill(active, active + size, true);
+        std::vector<UInt32> previous(size + 1);
+        std::vector<UInt32> next(size + 1);
+        std::vector<bool> active(size + 1, true);
```
Comment (davenger) [inline src/AggregateFunctions/AggregateFunctionHistogram.cpp:135]: Is it possible to use std::arrays instead of vectors here not to introduce allocations?

Comment (nickitat) [reply src/AggregateFunctions/AggregateFunctionHistogram.cpp:135]: i don't think so, since it'd also require size to be known at compile time

Comment (davenger) [reply src/AggregateFunctions/AggregateFunctionHistogram.cpp:135]: Of cause, you right

**Review by davenger — APPROVED** [review]

**Pull request merged**